### DISCUSSION TITLE
Issues with buildModelFMU function

### DISCUSSION THREAD

**Discussion by dietmarw** [OpenModelica/OpenModelica#10544] [Q&A] [answered]
I'm staring this as a discussion since I'm not sure if this is really an issue or just me not doing it right.

In connection with testing #10485 I've tried run this testscript to generate a source-code only FMU:

```mo
loadModel(Modelica);
getErrorString();
buildModelFMU(Modelica.Blocks.Sources.IntegerConstant, version="2.0", fmuType="cs",
              fileNamePrefix="ConstInteger", platforms={}, includeResources=false);
getErrorString();
```
Which according to https://build.openmodelica.org/Documentation/OpenModelica.Scripting.buildModelFMU.html should be correct.  However this will fail, because of the `platforms={}, includeResources=false` part (either of them will break the call). Now I was confused and double checked what OMEdit calls when generating a source-code-only FMU and its call is:

`buildModelFMU(Modelica.Blocks.Sources.IntegerConstant,"2.0","cs","ConstInteger",{},false)`

That call would also fail when run from a script (the OMEdit export succeeds). It will only ever succeed if I omit the last two arguments (which then will build a binary in addition to the source).

What am I doing wrong here?

**Comment by dietmarw** [comment]
One more thing that puzzles me is, why is there [translateModelFMU](https://build.openmodelica.org/Documentation/OpenModelica.Scripting.translateModelFMU.html) when it does the same as [buildModelFMU](https://build.openmodelica.org/Documentation/OpenModelica.Scripting.buildModelFMU.html) including **building** a binary which is even more confusing.

**Reply by perost** [reply]
Looking at the code `translateModelFMU` is identical to `buildModelFMU` except that it lacks the `platforms` parameter. So there doesn't really seem to be much point in `translateModelFMU` since any call to it could be replaced with `buildModelFMU`.

Also, both functions have a parameter `includeResources` which is just ignored by the compiler.

**Reply by dietmarw** [reply]
OK, in that case I should probably file an issue that `translateModelFMU` should become an alias for `buildModelFMU` (to not brake user scripts) but update the documentation with a warning that it is obsolete and might get removed in a future version of OM.

**Reply by AnHeuermann** [reply]
It's already an alias (sort of) of `buildModelFMU`.

**Reply by dietmarw** [reply]
So would it then not be a good idea to discourage people from using it? What about the "includeResources" bit, Should that be removed if it doesn't do anything?

**Reply by AnHeuermann** [reply]
Yes and yes (or fixed and the compiler flag removed).

**Reply by dietmarw** [reply]
OK I moved that part to #10550

**Comment by AnHeuermann** [comment, marked as answer]
Seems to be a bug in the OpenModelicaScriptingAPI. @kabdelhak is debugged it.

**Reply by dietmarw** [reply]
So should I convert this discussion to a ticket? Or is there already one for it?

**Reply by AnHeuermann** [reply]
Done https://github.com/OpenModelica/OpenModelica/issues/10554
